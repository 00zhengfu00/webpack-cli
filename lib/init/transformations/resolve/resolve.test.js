"use strict";

const defineTest = require("../../../utils/defineTest");

defineTest(
	__dirname,
	"resolve",
	"resolve-0",
	{
		alias: {
			inject: "{{#if_eq build 'standalone'}}",
			hello: "'world'",
			inject_1: "{{/if_eq}}",
			world: "hello"
		},
		aliasFields: ["'browser'", "wars"],
		descriptionFiles: ["'a'", "b"],
		enforceExtension: false,
		enforceModuleExtension: false,
		extensions: ["hey", "'ho'"],
		mainFields: ["main", "'story'"],
		mainFiles: ["'noMainFileHere'", "iGuess"],
		modules: ["one", "'two'"],
		unsafeCache: false,
		resolveLoader: {
			modules: ["'node_modules'", "mode_nodules"],
			extensions: ["jsVal", "'.json'"],
			mainFields: ["loader", "'main'"],
			moduleExtensions: ["'-loader'", "value"]
		},
		plugins: ["somePlugin", "'stringVal'"],
		symlinks: true
	},
	"init"
);

defineTest(
	__dirname,
	"resolve",
	"resolve-1",
	{
		alias: {
			inject: "{{#isdf_eq buildasda 'staasdndalone'}}",
			hello: "'worlasdd'",
			inject_1: "{{/asd}}",
			world: "asdc"
		},
		aliasFields: ["'as'"],
		descriptionFiles: ["'d'", "e", "f"],
		enforceExtension: true,
		extensions: ["ok", "'ho'"],
		mainFields: ["ok", "'story'"],
		mainFiles: ["'noMainFileHere'", "niGuess"],
		resolveLoader: {
			modules: ["'ok'", "mode_nodules"],
			mainFields: ["no", "'main'"],
			moduleExtensions: ["'-kn'", "ok"]
		},
		plugins: ["somePlugin", "'stringVal'"],
		symlinks: false
	},
	"add"
);
