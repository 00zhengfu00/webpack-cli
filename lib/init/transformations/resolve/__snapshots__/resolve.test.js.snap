// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolve transforms correctly using "resolve-0" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    resolve: {
        alias: {
            {{#if_eq build 'standalone'}},
            hello: 'world',
            {{/if_eq}},
            world: hello
        },

        aliasFields: ['browser', wars],
        descriptionFiles: ['a', b],
        enforceExtension: false,
        enforceModuleExtension: false,
        extensions: [hey, 'ho'],
        mainFields: [main, 'story'],
        mainFiles: ['noMainFileHere', iGuess],
        modules: [one, 'two'],
        unsafeCache: false,

        resolveLoader: {
            modules: ['node_modules', mode_nodules],
            extensions: [jsVal, '.json'],
            mainFields: [loader, 'main'],
            moduleExtensions: ['-loader', value]
        },

        plugins: [somePlugin, 'stringVal'],
        symlinks: true
    }
}
"
`;

exports[`resolve transforms correctly using "resolve-1" data 1`] = `
"module.exports = {
	entry: 'index.js',
	output: {
		filename: 'bundle.js'
	},
	resolve: {
        alias: {
            inject: {{#isdf_eq buildasda 'staasdndalone'}},
            hello: \\"'world'\\",
            inject_1: {{/asd}},
            world: \\"hello\\",
            hello: 'worlasdd',
            world: asdc
        },

        aliasFields: [\\"'browser'\\", \\"wars\\", 'browser', wars, 'as'],
        descriptionFiles: [\\"'a'\\", \\"b\\", \\"'c'\\", 'a', b, 'c', 'd', e, f],
        enforceExtension: false,
        enforceModuleExtension: false,
        extensions: [\\"hey\\", \\"ho\\", hey, ho, ok, 'ho'],
        mainFields: [\\"main\\", \\"'story'\\", main, 'story', ok, main, 'story', no, 'main'],
        mainFiles: [\\"'noMainFileHere'\\", \\"iGuess\\", 'noMainFileHere', iGuess, niGuess],
        modules: [\\"one\\", \\"'two'\\", one, 'two', 'ok', mode_nodules],
        unsafeCache: false,

        resolveLoader: {
			modules: [\\"'node_modules'\\", \\"mode_nodules\\", 'node_modules', mode_nodules, 'ok'],
			extensions: [\\"jsVal\\", \\"'.json'\\", jsVal, '.json', ok, 'ho'],
			mainFields: [\\"loader\\", \\"'main'\\", loader, 'main', ok, 'story', loader, 'main', no],
			moduleExtensions: [\\"'-loader'\\", \\"value\\", '-loader', value, '-kn', ok]
		},

        plugins: [\\"somePlugin\\", \\"'stringVal'\\", somePlugin, 'stringVal'],
        symlinks: true,
        enforceExtension: true,
        symlinks: false
    }
}
"
`;
